File: mobile/lib/models/govt_scheme.dart
class GovtSchemeInfo {
  final int id;
  final String title;
  final String category;
  final String region;
  final String description;
  final String eligibility;
  final String howToApply;
  final String? contactInfo;

  GovtSchemeInfo({
    required this.id,
    required this.title,
    required this.category,
    required this.region,
    required this.description,
    required this.eligibility,
    required this.howToApply,
    this.contactInfo,
  });

  factory GovtSchemeInfo.fromJson(Map<String, dynamic> json) => GovtSchemeInfo(
        id: json['id'] as int,
        title: json['title'] as String,
        category: json['category'] as String,
        region: json['region'] as String,
        description: json['description'] as String,
        eligibility: json['eligibility'] as String,
        howToApply: json['how_to_apply'] as String,
        contactInfo: json['contact_info'] as String?,
      );

  Map<String, dynamic> toCacheJson() => {
        'id': id,
        'title': title,
        'category': category,
        'region': region,
        'description': description,
        'eligibility': eligibility,
        'how_to_apply': howToApply,
        'contact_info': contactInfo,
      };
}

File: backend/app/models/govt_scheme.py
"""Government welfare / subsidy schemes for fishermen, used by the Schemes screen."""
from sqlalchemy import Column, Integer, String, Text, Boolean, DateTime, func
from app.database import Base


class GovtScheme(Base):
    __tablename__ = "govt_schemes"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(200), nullable=False)
    category = Column(String(60), nullable=False)  # subsidy, insurance, loan, relief, training
    region = Column(String(120), nullable=False, default="National")
    description = Column(Text, nullable=False)
    eligibility = Column(Text, nullable=False)
    how_to_apply = Column(Text, nullable=False)
    contact_info = Column(String(255), nullable=True)
    is_active = Column(Boolean, default=True, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
